client/rest-high-level/src/main/java/org/elasticsearch/client/RestHighLevelClient.java

```diff
@@ -1011,9 +1012,9 @@ public final void fieldCapsAsync(FieldCapabilitiesRequest fieldCapabilitiesReque
                                                              RequestOptions options,
                                                              CheckedFunction<Response, Resp, IOException> responseConverter,
                                                              Set<Integer> ignores) throws IOException {
-        ValidationException validationException = request.validate();
-        if (validationException != null && validationException.validationErrors().isEmpty() == false) {
-            throw validationException;
+        Optional<ValidationException> validationException = request.validate();
+        if (validationException != null && validationException.isPresent()) {
```

Comment:
The philosophy I've usually gone with is that you shouldn't null-check `Optional`s - just let it throw an NPE because you use an empty Optional to represent "no value" so a null value means something is misbehaving.

The counterargument is that leaving the null-check here is probably safer, especially if we're converting code that *used to* use null to represent "no value" - but if we're going to go with that logic, we should probably still check for the `isEmpty()` case too.

Reply:
* we used to the null check to mean "no validation errors", therefore I am using `isPresent()`. My only fear about removing the null check could trip people because they might still return null, instead of `Optional.EMPTY` or `Optional.ofNullable`. 

I think the `isEmpty()` is less of an issue than before, given that we have to redo the code here anyway. Its just a matter of accounting for the null check itself. I think its worthwhile to remove the null check once we have moved all the actions over, but to remove the `isEmpty()` check now to reduce a potential bug (any non null validation exception will be assumed to be a valid exception). Thoughts?

Reply:
Both used to mean "no validation errors", but looking through the code it looks like returning a request where `isEmpty()` is true is pretty rare.  So I'm okay with removing that check now and removing the null check later.